From: Christian Kastner <ckk@kvr.at>
Date: Sun, 13 Apr 2025 12:07:10 +0200
Subject: test: Don't rely on PAGE_SIZE for maxdesc length

The maxdesc length was fixed at 4096 in kernel 3.18 rather than
PAGE_SIZE, and toolbox.inc.sh accounts for that when setting $maxdesc.
Rather than introducing yet another variable, calculate the length of
$maxdesc at runtime in the few spots where this is needed.

This was discovered on a POWER9 host with 64K page size when running
tests as non-root. Because 64K page size < 20000 quota on this host,
the else branch was entered, where the expected EDQUOT did not occur
because the hard 4096 $maxdesc introduced in 3.18 was of course
within the 20000 quota.
---
 tests/keyctl/add/bad-args/runtest.sh     | 4 ++--
 tests/keyctl/newring/bad-args/runtest.sh | 4 ++--
 tests/keyctl/padd/bad-args/runtest.sh    | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/tests/keyctl/add/bad-args/runtest.sh b/tests/keyctl/add/bad-args/runtest.sh
index 69121b2..d7c3405 100644
--- a/tests/keyctl/add/bad-args/runtest.sh
+++ b/tests/keyctl/add/bad-args/runtest.sh
@@ -39,8 +39,8 @@ marker "CHECK ADD KEYRING WITH PAYLOAD"
 create_key --fail keyring wibble a @p
 expect_error EINVAL
 
-# check that an max length key description works correctly (PAGE_SIZE inc NUL)
-if [ $PAGE_SIZE -lt $maxsquota ]
+# check that an max length key description works correctly
+if [ `expr length $maxdesc` -lt $maxsquota ]
 then
     marker "CHECK MAXLEN DESC"
     create_key --new=keyid user $maxdesc stuff @s
diff --git a/tests/keyctl/newring/bad-args/runtest.sh b/tests/keyctl/newring/bad-args/runtest.sh
index f35c4c6..b2df70d 100644
--- a/tests/keyctl/newring/bad-args/runtest.sh
+++ b/tests/keyctl/newring/bad-args/runtest.sh
@@ -9,8 +9,8 @@
 result=PASS
 echo "++++ BEGINNING TEST" >$OUTPUTFILE
 
-# check that an max length key description works correctly (4096 inc NUL)
-if [ $PAGE_SIZE -lt $maxsquota ]
+# check that an max length key description works correctly
+if [ `expr length $maxdesc` -lt $maxsquota ]
 then
     marker "CHECK MAXLEN DESC"
     create_keyring --new=keyid $maxdesc @s
diff --git a/tests/keyctl/padd/bad-args/runtest.sh b/tests/keyctl/padd/bad-args/runtest.sh
index c59393c..fb5e4f4 100644
--- a/tests/keyctl/padd/bad-args/runtest.sh
+++ b/tests/keyctl/padd/bad-args/runtest.sh
@@ -40,7 +40,7 @@ pcreate_key --fail stuff keyring wibble @p
 expect_error EINVAL
 
 # check that an max length key description works correctly
-if [ $PAGE_SIZE -lt $maxsquota ]
+if [ `expr length $maxdesc` -lt $maxsquota ]
 then
     marker "CHECK MAXLEN DESC"
     pcreate_key --new=keyid stuff user $maxdesc @s
